### PR TITLE
Use new serializer in contract

### PR DESCRIPTION
**Pull request by Solpatium**

### CODE REVIEW

File: starknet_py/contract.py [hunk 1]
```diff
@@ -45,14 +52,23 @@
...
+    @cached_property
+    def identifier_manager(self) -> IdentifierManager:
+        warnings.warn(
+            "Using identifier_manager from ContractData has been deprecated. Consider using parsed_abi instead."
+        )
+        return identifier_manager_from_abi(self.abi)
+
+    @cached_property
+    def parsed_abi(self) -> Abi:
+        return AbiParser(self.abi).parse()
```
Comment (cptartur) [inline starknet_py/contract.py:65]: Is it documented somewhere?

Comment (cptartur) [reply starknet_py/contract.py:65]: Maybe a docstring would be good here?

Comment (Solpatium) [reply starknet_py/contract.py:65]: This is basically a property of the object, but since it is frozen it is easier to do it this way.

Comment (Solpatium) [reply starknet_py/contract.py:65]: I can add docstring

File: starknet_py/cairo/serialization/data_serializers/array_serializer_test.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import pytest
+from starkware.crypto.signature.signature import FIELD_PRIME
 
-from starknet_py.cairo.felt import FIELD_PRIME
```
Comment (war-in) [inline starknet_py/cairo/serialization/data_serializers/array_serializer_test.py:3]: But why? 🥺

**Review by war-in — APPROVED** [review]

File: starknet_py/cairo/serialization/data_serializers/array_serializer_test.py (hunk 2 above)
Comment (Solpatium) [reply starknet_py/cairo/serialization/data_serializers/array_serializer_test.py:3]: Because it is just imported there.

**Pull request merged**